scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -213,33 +214,69 @@ class ExperimentalScanner(
+                                log.debug {
+                                    "Skipping scan of '${pkg.id.toCoordinates()}' with package scanner " +
+                                            "'${scanner.name}' as stored results are available."
+                                }
+                            }
+                        }
                     }
 
-                    return@scanner
+                if (packagesWithIncompleteScanResult.isEmpty()) return@scanner
+
+                // Create a reference package with any VCS path removed, to ensure the full repository is scanned.
+                val referencePackage = packagesWithIncompleteScanResult.first().let { pkg ->
+                    if (provenance is RepositoryProvenance) {
+                        pkg.copy(vcsProcessed = pkg.vcsProcessed.copy(path = ""))
+                    } else {
+                        pkg
+                    }
                 }
 
-                log.info { "Scan of '${pkg.id.toCoordinates()}' with package scanner '${scanner.name}' started." }
+                if (packagesWithIncompleteScanResult.size > 1) {
+                    log.info {
+                        val packageIds = packagesWithIncompleteScanResult.drop(1)
+                            .joinToString("\n") { "\t${it.id.toCoordinates()}" }
+                        "Scanning package '${referencePackage.id.toCoordinates()}' as reference for these packages " +
+                                "with the same provenance:\n$packageIds"
+                    }
+                }
 
-                val scanResult = scanner.scanPackage(pkg, context)
+                log.info {
+                    "Scan of '${referencePackage.id.toCoordinates()}' with package scanner '${scanner.name} started."
+                }
 
-                log.info { "Scan of '${pkg.id.toCoordinates()}' with package scanner '${scanner.name}' finished." }
+                val scanResult = scanner.scanPackage(referencePackage, context)
 
-                val nestedProvenanceScanResult = scanResult.toNestedProvenanceScanResult(nestedProvenance)
+                log.info {
+                    "Scan of '${referencePackage.id.toCoordinates()}' with package scanner '${scanner.name}' finished."
+                }
 
-                controller.addNestedScanResult(scanner, nestedProvenanceScanResult)
+                packagesWithIncompleteScanResult.forEach processResults@{ pkg ->
+                    val nestedProvenance = controller.getNestedProvenance(pkg.id) ?: return@processResults
+                    val nestedProvenanceScanResult = scanResult.toNestedProvenanceScanResult(nestedProvenance)
+                    controller.addNestedScanResult(scanner, nestedProvenanceScanResult)
 
-                // TODO: Run in coroutine.
-                if (scanner.criteria != null) {
-                    storeNestedScanResult(pkg, nestedProvenanceScanResult)
+                    if (scanner.criteria != null) {
```

Comment:
Is the TODO to run in a coroutine obsolete now?

Reply:
Recovered.